doc/guides/test/ztest.rst

```diff
@@ -377,6 +377,59 @@ Other questions:
   themselves -- only one file to update when changes are made
   eliminates duplication.
 
+Stress test framework
+*********************
+
+Zephyr stress test framework (Ztress) provides environment for executing user
+functions in multiple priority contexts. It can be used to validate that code is
+resilient to preemptions. Framework tracks number of executions and preemptions
+for each context. Execution can have various completion conditions like timeout,
+number of executions or number of preemptions.
+
+Framework is setting up the environment by creating requested number of threads
+(each on different priority), optionally starting a timer. For each context, user
+function (different for each context) is called and then context sleeps for
+randomized amount of system ticks. Framework is tracking CPU load and adjusts sleeping
+periods to achieve higher CPU load. In order to increase probability of preemptions,
+system clock frequency should be relatively high. Default 100 Hz on QEMU x86
+is much too low and it is recommended to increase it to 100 kHz.
+
+Stress test environment is setup and run using :c:macro:`ZTRESS_EXECUTE` which
+accepts variable number of arguments. Each argument is a context which is
+specified by :c:macro:`ZTRESS_TIMER` or :c:macro:`ZTRESS_THREAD` macros. Context
+are specified in priority descending order. Each context specifies completion
+conditions by providing minimum number of executions and preemptions. When all
+conditions are met, run is completed, execution report is printed and macro returns.
+When test is executed, progress report is periodically printed.
```

Comment:
conditions are met and the execution has completed, an execution report is printed and the macro returns. Note that while the test is executing, a progress report is periodically printed.